pkg/frontend/querymiddleware/codec.go

```diff
@@ -311,12 +313,18 @@ func (prometheusCodec) DecodeLabelsQueryRequest(_ context.Context, r *http.Reque
 
 	labelMatcherSets := reqValues["match[]"]
 
+	limit, err := strconv.ParseInt(reqValues.Get("limit"), 10, 64)
+	if reqValues.Has("limit") && err != nil {
+		return nil, apierror.New(apierror.TypeBadData, "limit parameter: "+err.Error())
+	}
+
 	if IsLabelNamesQuery(r.URL.Path) {
 		return &PrometheusLabelNamesQueryRequest{
 			Path:             r.URL.Path,
 			Start:            start,
 			End:              end,
 			LabelMatcherSets: labelMatcherSets,
+			Limit:            limit,
```

Comment:
IIUC, passing the limit value no matter what means we rely on whatever `ParseInt()` returns in the error case when there's no limit. Would it be better (more obvious) to set a default value above and conditionally call `ParseInt()`?

Reply:
Makes sense.

this made me realize that I had missed setting the `limit` parameter on the HTTP request which the query-frontend creates (after processing the protobuf model). This wasn't a problem because the protobufs were only being used in caching logic (so no encoding into HTTP happens). Fixed in the last commit https://github.com/grafana/mimir/pull/7722/commits/663121cb394ca9ba79a906589e3a2bfef7218b65